libindy/indy-utils/src/crypto/mod.rs

```diff
@@ -38,7 +47,315 @@ pub mod xsalsa20;
 #[path = "ed25519_sign/sodium.rs"]
 pub mod ed25519_sign;
 
+pub mod pack;
+
 #[cfg(feature = "ed25519_box_sodium")]
 #[path = "ed25519_box/sodium.rs"]
 // TODO: The name is misleading as the operations do not happen over ed25519 curve
-pub mod ed25519_box;
\ No newline at end of file
+pub mod ed25519_box;
+
+pub mod verkey_builder;
+
+use indy_api_types::errors::prelude::*;
+use verkey_builder::{DEFAULT_CRYPTO_TYPE, clear_verkey};
+
+pub fn create_key(key_info: &KeyInfo) -> IndyResult<Key> {
+    trace!("create_key >>> key_info: {:?}", secret!(key_info));
+
+    match key_info.crypto_type.as_ref().map(String::as_str) {
+        Some(DEFAULT_CRYPTO_TYPE) | None => {}
```

Comment:
Using DEFAULT_CRYPTO_TYPE is only correct as long as the default is ed25519.

Reply:
I added this `match` to keep old behavior. else we get other error in this case.